New Diagnostics tool

gingerwizard opened the pull request:
### Changelog category
- New Feature

### Changelog entry
New single binary based diagnostics tool.

gingerwizard committed:
Initial commit

alexey-milovidov commented:
> [tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log.gz](https://github.com/ClickHouse/ClickHouse/pull/36705/files#diff-5785673a226a9aa79229765e914b7f6052864ea920daef75c8ad1225e2a735b7)

Is it needed?

alexey-milovidov commented on tools/clickhouse-diagnostics/CONTRIBUTION.md:
````diff
@@ -0,0 +1,92 @@
+2. Create a new tag with a higher version number e.g.
+
+```bash
+git tag -a v1.0.0-beta.3 -m "A summary comment"
````
This becomes misleading.

alexey-milovidov commented on tools/clickhouse-diagnostics/CONTRIBUTION.md:
````diff
@@ -0,0 +1,92 @@
+5. Run a release
+
+```bash
+make release
````
But there is no makefile.

gingerwizard replied:
will add.

alexey-milovidov commented on tools/clickhouse-diagnostics/CONTRIBUTION.md:
```diff
@@ -0,0 +1,92 @@
+# Contribution
+
+We keep things simple. Execute all commands in the root folder.
```
root folder - "this folder"

alexey-milovidov commented:
Let's also correct this: https://s3.amazonaws.com/clickhouse-test-reports/36705/f8dfe1906f9d19bef4855de0f51df2455dd318f9/style_check__actions_/style_output.txt

gingerwizard committed:
Response to comments

gingerwizard commented:
> Let's also correct this: https://s3.amazonaws.com/clickhouse-test-reports/36705/f8dfe1906f9d19bef4855de0f51df2455dd318f9/style_check__actions_/style_output.txt

Thanks go-lint doesnt check the yaml.

> [tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log.gz](https://github.com/ClickHouse/ClickHouse/pull/36705/files#diff-5785673a226a9aa79229765e914b7f6052864ea920daef75c8ad1225e2a735b7)
> 
> Is it needed?
> 

Its part of the tests to confirm gz logs are copied. I removed the file and replaced with a dummy file with an indicator as such i.e. its not just a text file to minimize space.

alexey-milovidov commented:
One minor thing: https://s3.amazonaws.com/clickhouse-test-reports/36705/1c0e258c8bbb083bbfa751d9dc87ebefd2633d82/style_check__actions_/style_output.txt

gingerwizard committed:
Style fixes

gingerwizard commented:
How to run this lint check locally over just a sub folder? I see build executes

```docker run -u $(id -u ${USER}):$(id -g ${USER}) --cap-add=SYS_PTRACE --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse:/ClickHouse --volume=/home/ubuntu/actions-runner/_work/_temp/style_check:/test_output clickhouse/style-test:latest```

Is there a simpler way to do locally?
